doc/api/quic.md

```diff
@@ -0,0 +1,1958 @@
+by the `alpn` configuration option. In QUIC Applications that support headers,
+informational header blocks typically come before initial headers.
+
+The event handler is invoked with a single argument representing the block of
+Headers as an object.
+
+```js
+stream('informationalHeaders', (headers) => {
+  // Use headers
+});
+```
+
+#### Event: `'initialHeaders'`
+<!-- YAML
+added: REPLACEME
+-->
+
+Emitted when the `QuicStream` has received a block of initial headers.
+
+Support for headers depends entirely on the QUIC Application used as identified
+by the `alpn` configuration option. HTTP/3, for instance, supports two kinds of
+initial headers: request headers for HTTP request messages and response headers
+for HTTP response messages. For HTTP/3 QUIC streams, request and response
+headers are each emitted using the `'initialHeaders'` event.
+
+The event handler is invoked with a single argument representing the block of
+Headers as an object.
+
+```js
+stream('initialHeaders', (headers) => {
+  // Use headers
+});
+```
+
+#### Event: `'ready'`
+<!-- YAML
+added: REPLACEME
+-->
+
+Emitted when the underlying `QuicSession` has emitted its `secure` event
+this stream has received its id, which is accessible as `stream.id` once this
+event is emitted.
+
+#### Event: `'trailingHeaders'`
+<!-- YAML
+added: REPLACEME
+-->
+
+Emitted when the `QuicStream` has received a block of trailing headers.
+
+Support for headers depends entirely on the QUIC Application used as identified
+by the `alpn` configuration option. Trailing headers typically follow any data
+transmitted on the `QuicStream`, and therefore typically emit sometime after the
+last `'data'` event but before the `'close'` event. The precise timing may
+vary from one QUIC application to another.
+
+The event handler is invoked with a single argument representing the block of
+Headers as an object.
+
+```js
+stream('trailingHeaders', (headers) => {
+  // Use headers
+});
+```
+
+#### Event: `'readable'`
+<!-- YAML
+added: REPLACEME
+-->
+
+#### quicstream.aborted
+<!-- YAML
+added: REPLACEME
+-->
+* Type: {boolean}
+
+True if dataflow on the `QuicStream` was prematurely terminated.
+
+#### quicstream.bidirectional
+<!--YAML
+added: REPLACEME
+-->
+
+* Type: {boolean}
+
+Set to `true` if the `QuicStream` is bidirectional.
+
+#### quicstream.bytesReceived
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {BigInt}
+
+A `BigInt` representing the total number of bytes received for this
+`QuicStream`.
+
+#### quicstream.bytesSent
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {BigInt}
+
+A `BigInt` representing the total number of bytes sent by this
+`QuicStream`.
+
+#### quicstream.clientInitiated
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {boolean}
+
+Set to `true` if the `QuicStream` was initiated by a `QuicClientSession`
+instance.
+
+#### quicstream.close(code)
+<!-- YAML
+added: REPLACEME
+-->
+
+* `code` {number}
+
+Closes the `QuicStream`.
```

Comment:
What's the difference between `QuicStream.close` and `Duplex.destroy`?

Reply:
Seems like a graceful variant of `destroy`. Is this something we should move one level up into `Duplex`. Graceful shutdown of `Readable` is `push(null)` and for `Writable` it is `end()`, but I guess this is for `Duplex` which should do both.

Reply:
We could potentially add a similar concept to `Duplex` in general. Would have to see what makes sense.

Reply:
Just realized that I hadn't fully answered this: QuicStream.close is a graceful operation. It allows any data in flight to be fully handled without allowing additional data to be written. Destroy, on the otherhand, immediately frees all resources and cancels pending writes.